hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/balancer/BalancingPolicy.java

```diff
@@ -122,6 +123,45 @@ Double getUtilization(DatanodeStorageReport r, final StorageType t) {
     }
   }
 
+  /**
+   * Cluster is balanced if each node is balanced. Unlike {@link Node},
+   * The policy take into account non-DFS used spaces.
+   * <p>
+   * Actual DFS capacity (capacity - non-DFS used) is calculated by
```

Comment:
Intuitively, I would have expected the code to calculate this as `s.getCapacity() - s.getNonDfsUsed()`. Is there some reason this calculation doesn't work and instead we need to infer it as (DFS used + remaining)?